openedx/features/badging/models.py

```diff
@@ -119,58 +134,45 @@ def assign_badge(cls, user_id, badge_id, community_id):
 
             if not team_group_chat:
                 error = badge_constants.INVALID_TEAM_ERROR.format(badge_id=badge_id, community_id=community_id)
-                log.exception(error)
                 raise Exception(error)
 
             course_id = team_group_chat['team__course_id']
 
             if not course_id or course_id == CourseKeyField.Empty:
                 error = badge_constants.UNKNOWN_COURSE_ERROR.format(badge_id=badge_id, community_id=community_id)
-                log.exception(error)
                 raise Exception(error)
 
             all_team_members = CourseTeamMembership.objects.filter(team_id=team_group_chat['team_id'])
             assigned = False
             for member in all_team_members:
-                user_badge, assigned = UserBadge.objects.get_or_create(
-                    user_id=member.user_id,
-                    badge_id=badge_id,
-                    course_id=course_id,
-                    community_id=community_id
-                )
+                # assign badge to all team members, also send email and notification to all team members
+                _, assigned = create_user_badge_and_notify(member.user_id, badge_id, course_id, community_id)
 
             return assigned
         elif badge_type == badge_constants.CONVERSATIONALIST[CONVERSATIONALIST_ENTRY_INDEX]:
             course_id = get_course_id_by_community_id(community_id)
 
             if not course_id or course_id == CourseKeyField.Empty:
                 error = badge_constants.INVALID_COMMUNITY_ERROR.format(badge_id=badge_id, community_id=community_id)
-                log.exception(error)
                 raise Exception(error)
 
-            user_badge, assigned = UserBadge.objects.get_or_create(
-                user_id=user_id,
-                badge_id=badge_id,
-                course_id=course_id,
-                community_id=community_id
-            )
+            # assign badge to current user, also send email and notification
+            _, assigned = create_user_badge_and_notify(user_id, badge_id, course_id, community_id)
```

Comment:
since we're not using the first returned value of this internal method anywhere maybe we can only return the assigned flag